plugins/module_utils/foreman_helper.py

```diff
@@ -575,8 +482,15 @@ def _entity_spec_helper(spec):
         elif argument_value.get('type') == 'entity_list':
             argument_value['type'] = 'list'
             entity_value['type'] = 'entity_list'
-        entity_spec[key] = entity_value
-        argument_spec[key] = argument_value
+        elif argument_value.get('type') == 'nested_list':
+            argument_value['type'] = 'list'
+            argument_value['elements'] = 'dict'
+            argument_value['options'] = _entity_spec_helper(argument_value.pop('entity_spec'))[1]
```

Comment:
```suggestion
            _, argument_value['options'] = _entity_spec_helper(argument_value.pop('entity_spec'))
```

Up to you, but I find `_` more readable than `[1]` at the end.